Docs: Document nested tables and nested field overrides in the table visualization page (#124425)
grafana/grafana: docs/sources/visualizations/panels-visualizations/visualizations/table/index.md

      destination: /docs/grafana/<GRAFANA_VERSION>/panels-visualizations/visualizations/time-series/#graph-styles-options
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana-cloud/visualizations/panels-visualizations/visualizations/time-series/#graph-styles-options
  group-to-nested-tables-transformation:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/panels-visualizations/query-transform-data/transform-data/#group-to-nested-tables
    - pattern: /docs/grafana-cloud/

>       destination: /docs/grafana-cloud/visualizations/panels-visualizations/query-transform-data/transform-data/#group-to-nested-tables

Reviewer: ```suggestion
```

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-multi-dataset-v11.3.png" max-width="650px" alt="Table visualization with multiple datasets" >}}

## Nested tables

A table can display sub-tables inside expandable rows. You can add these nested tables using the [Group to nested tables transformation](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/panels-visualizations/query-transform-data/transform-data/#group-to-nested-tables), which groups rows by one or more fields, and can summarize nested row data by applying calculations.

> 

Reviewer: ```suggestion

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-collapsed-v13.1.png" max-width="750px" alt="Table with all rows collapsed" >}}

```

Click the expand icon on a row to toggle the visibility of its nested table:

> {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-nested-expanded-v13.x.png" max-width="750px" alt="Table with one row expanded showing a nested sub-table" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-expanded-v13.1.png" max-width="750px" alt="Table with two rows expanded showing nested sub-tables" >}}
```

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-multi-dataset-v11.3.png" max-width="650px" alt="Table visualization with multiple datasets" >}}

## Nested tables

A table can display sub-tables inside expandable rows. You can add these nested tables using the [Group to nested tables transformation](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/panels-visualizations/query-transform-data/transform-data/#group-to-nested-tables), which groups rows by one or more fields, and can summarize nested row data by applying calculations.

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-collapsed-v13.1.png" max-width="750px" alt="Table with all rows collapsed" >}}

Click the expand icon on a row to toggle the visibility of its nested table:

> {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-expanded-v13.1.png" max-width="750px" alt="Table with two rows expanded showing nested sub-tables" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-collapsed-v13.1.png" max-width="650px" alt="Table with all rows collapsed" >}}

Click the expand icon on a row to toggle the visibility of its nested table:

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-expanded-v13.1.png" max-width="650px" alt="Table with two rows expanded showing nested sub-tables" >}}
```

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-multi-dataset-v11.3.png" max-width="650px" alt="Table visualization with multiple datasets" >}}

## Nested tables

A table can display sub-tables inside expandable rows.

> Nested tables are produced by the [Group to nested tables transformation](ref:group-to-nested-tables-transformation), which groups rows by one or more fields, and can summarize nested row data by applying calculations.

Reviewer: We're slowly ripping out all the ref URIs and replacing them with regular URLs. The tooling will infer the Cloud link.

```suggestion
Nested tables are produced by the [Group to nested tables transformation](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/panels-visualizations/query-transform-data/transform-data/#group-to-nested-tables), which groups rows by one or more fields, and can summarize nested row data by applying calculations.
```

> Nested tables are produced by the [Group to nested tables transformation](ref:group-to-nested-tables-transformation), which groups rows by one or more fields, and can summarize nested row data by applying calculations.

Reviewer: ```suggestion
A table can display sub-tables inside expandable rows.
You can add these nested tables using the [Group to nested tables transformation](ref:group-to-nested-tables-transformation), which groups rows by one or more fields, and can summarize nested row data by applying calculations.
```

Click the expand icon on a row to toggle the visibility of its nested table:

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-nested-expanded-v13.x.png" max-width="750px" alt="Table with one row expanded showing a nested sub-table" >}}

> Sorting nested and top-level rows in a nested table uses the same interaction as sorting multiple columns in a flat table - refer to [Sort columns](#sort-columns) to learn more.

Reviewer: This isn't a lot of text, so I think we should just repeat it rather than making the reader click around. I've repeated the screenshot, but it might be worth taking a screenshot of a nested table to show this in that context.

```suggestion
To sort nested and top-level rows in nested tables, click a column title to change the sort order from default to descending to ascending.
Each time you click, the sort order changes to the next option in the cycle.
You can sort multiple columns by holding the `Cmd` or `Ctrl` key and clicking the column name.

{{< figure src="/static/img/docs/tables/sort-descending.png" max-width="350px" alt="Sort descending" class="docs-image--no-shadow" >}}
```

### Field overrides

{{< docs/shared lookup="visualizations/overrides-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}

#### Apply overrides to nested table fields

By default, field overrides apply only to columns in the parent table.
To target columns inside a nested table, set the **Target fields** option on the override to **Nested**:

> {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-override-nested-scope-v13.x.png" max-width="500px" alt="Field override configuration with the Target fields selector showing Series and Nested options" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-override-nested-scope-v13.x.png" max-width="350px" alt="Field override configuration with the Target fields selector showing Series and Nested options" >}}
```

All standard override properties&mdash;including thresholds, value mappings, units, data links, and cell type&mdash;apply the same way to nested fields.

> {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-nested-override-applied-v13.x.png" max-width="750px" alt="Nested table with a threshold override applied to a column inside an expanded sub-table" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-w-overrides-v13.1.png" max-width="750px" alt="Nested table with a threshold override applied to a column inside an expanded sub-table" >}}
```

### Field overrides

{{< docs/shared lookup="visualizations/overrides-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}

#### Apply overrides to nested table fields

By default, field overrides apply only to columns in the parent table.
To target columns inside a nested table, set the **Target fields** option on the override to **Nested**:

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-table-override-nested-scope-v13.x.png" max-width="350px" alt="Field override configuration with the Target fields selector showing Series and Nested options" >}}

All standard override properties&mdash;including thresholds, value mappings, units, data links, and cell type&mdash;apply the same way to nested fields.

> {{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-w-overrides-v13.1.png" max-width="750px" alt="Nested table with a threshold override applied to a column inside an expanded sub-table" >}}

Reviewer: ```suggestion
{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-nested-table-w-overrides-v13.1.png" max-width="650px" alt="Nested table with a threshold override applied to a column inside an expanded sub-table" >}}
```

### Field overrides

{{< docs/shared lookup="visualizations/overrides-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}

#### Apply overrides to nested table fields

{{< admonition type="note" >}}
This feature is in [public preview](https://grafana.com/docs/release-life-cycle/) and requires the `nestedFramesFieldOverrides` feature toggle.

> {{< /admonition >}}

Reviewer: ```suggestion
{{< docs/public-preview product="Nested table overrides" featureFlag="`nestedFramesFieldOverrides`" >}}
```
